packages/web3-providers-ws/src/index.ts

```diff
@@ -0,0 +1,387 @@
+			this.requestQueue.delete(id);
+			promise = requestItem.deferredPromise.realPromise as Promise<T>;
+		} else {
+			const defPromise = new DeferredPromise<T>();
+
+			const reqItem: WSRequestItem<T2, T> = {
+				payload: request,
+				deferredPromise: defPromise,
+			};
+
+			this.sentQueue.set(id, reqItem);
+			promise = defPromise.realPromise;
+		}
+
+		try {
+			this.webSocketConnection.send(JSON.stringify(request));
+		} catch (error) {
+			this.sentQueue.delete(id);
+			throw error;
+		}
+
+		return promise;
+	}
+
+	public removeAllListeners(type: string): void {
+		this.wsEventEmitter.removeAllListeners(type);
+	}
+
+	private init() {
+		this.lastDataChunk = '';
+		this.reconnectAttempts = 0;
+
+		if (this.lastChunkTimeout) {
+			clearTimeout(this.lastChunkTimeout);
+		}
+	}
+
+	private addSocketListeners(): void {
+		// eslint-disable-next-line
+		(this.webSocketConnection as any).addEventListener('message', this.onMessage.bind(this));
+		// eslint-disable-next-line
+		(this.webSocketConnection as any).addEventListener('open', this.onConnect.bind(this));
+		// eslint-disable-next-line
+		(this.webSocketConnection as any).addEventListener('close', this.onClose.bind(this));
+	}
+
+	private reconnect(): void {
+		this.providerStatus = 'connecting';
+
+		if (this.sentQueue.size > 0) {
+			this.sentQueue.forEach((request: WSRequestItem, key: JsonRpcId) => {
+				request.deferredPromise.reject(new PendingRequestsOnReconnectingError());
+				this.sentQueue.delete(key);
+			});
+		}
+
+		if (
+			!this.wsProviderOptions?.reconnectOptions?.maxAttempts ||
+			this.reconnectAttempts < this.wsProviderOptions.reconnectOptions.maxAttempts
+		) {
+			setTimeout(() => {
+				this.reconnectAttempts += 1;
+
+				this.removeSocketListeners();
+				this.connect();
+			}, this.wsProviderOptions?.reconnectOptions?.delay);
+		}
+	}
+
+	private onMessage(e: IMessageEvent): void {
+		this.parseResponse(typeof e.data === 'string' ? e.data : '').forEach(
+			(response: JsonRpcResponse | SubscriptionResultNotification) => {
+				if ('method' in response && response.method.endsWith('_subscription')) {
+					this.wsEventEmitter.emit('message', null, response);
+					return;
+				}
+
+				const { id } = response;
+
+				if (id && this.sentQueue.has(id)) {
+					const requestItem = this.sentQueue.get(id);
+					if ('result' in response && response.result !== undefined) {
+						this.wsEventEmitter.emit('message', null, response);
+						requestItem?.deferredPromise.resolve(response);
+					} else if ('error' in response && response.error !== undefined) {
+						this.wsEventEmitter.emit('message', response, null);
+						requestItem?.deferredPromise.reject(response);
+					}
+
+					this.sentQueue.delete(id);
+				}
+			},
+		);
+	}
+
+	private onConnect() {
+		this.providerStatus = 'connected';
+		this.reconnectAttempts = 0;
+
+		if (this.requestQueue.size > 0) {
+			for (const value of this.requestQueue.values()) {
+				// eslint-disable-next-line
```

Comment:
I think an explanation (like error message + reasons why it's not being fixed) should be provided when disabling linting. Additionally, you should be able to disable the specific rule for the line instead of disabling linting entirely